pkg/cache/shares.go

```diff
@@ -92,43 +92,31 @@ func AddShare(share *sharev1alpha1.Share) {
 	}
 }
 
-func UpdateShare(share *sharev1alpha1.Share) {
-	klog.V(4).Infof("UpdateShare key %s kind %s", share.Name, share.Spec.BackingResource.Kind)
+func UpdateShare(share *storagev1alpha1.SharedResource) {
+	klog.V(4).Infof("UpdateShare key %s type %s", share.Name, share.Spec.Resource.Type)
 	old, ok := shares.Load(share.Name)
 	if !ok || old == nil {
 		AddShare(share)
 		return
 	}
-	oldShare := old.(*sharev1alpha1.Share)
-	diffInstance := false
-	oldBr := oldShare.Spec.BackingResource
-	newBr := share.Spec.BackingResource
-	switch {
-	case oldBr.Kind != newBr.Kind:
-		diffInstance = true
-	case oldBr.Namespace != newBr.Namespace:
-		diffInstance = true
-	case oldBr.Name != newBr.Name:
-		diffInstance = true
-	}
-	klog.V(4).Infof("UpdateShare key %s kind %s diff %v", share.Name, share.Spec.BackingResource.Kind, diffInstance)
-	if !diffInstance {
+	oldShare := old.(*storagev1alpha1.SharedResource)
+
+	klog.V(4).Infof("UpdateShare key %s type %s", share.Name, share.Spec.Resource.Type)
+	if !reflect.DeepEqual(oldShare, share) {
```

Comment:
yeah I think the old way was legacy from when we used k8s OwnerReference 

nice this can be simplified